platform/security/handler/security-handler-api/src/main/java/org/codice/ddf/security/handler/api/BaseAuthenticationTokenFactory.java

```diff
@@ -0,0 +1,225 @@
+              null);
+    } catch (WSSecurityException | IOException e) {
+      LOGGER.warn("Unable to read merlin properties file. Unable to validate certificates.", e);
+    }
+    Init.init();
+  }
+
+  /**
+   * Creates a {@link BaseAuthenticationToken} from a given username and password. Uses a {@link
+   * UsernameTokenType} internally to store the username and password.
+   *
+   * @param username - user's username
+   * @param password - user's password
+   * @return a BaseAuthenticationToken containing the given username and password
+   */
+  public BaseAuthenticationToken getTokenFromUsernamePassword(String username, String password) {
+    UsernameTokenType usernameTokenType = new UsernameTokenType();
+    AttributedString user = new AttributedString();
+    user.setValue(username);
+    usernameTokenType.setUsername(user);
+    String usernameToken = null;
+
+    // Add a password
+    PasswordString pass = new PasswordString();
+    pass.setValue(password);
+    pass.setType(WSConstants.PASSWORD_TEXT);
+    JAXBElement<PasswordString> passwordType =
+        new JAXBElement<>(QNameConstants.PASSWORD, PasswordString.class, pass);
+    usernameTokenType.getAny().add(passwordType);
+
+    // Marshall the received JAXB object into a DOM Element
+    List<String> ctxPath = new ArrayList<>(2);
+    ctxPath.add(ObjectFactory.class.getPackage().getName());
+    ctxPath.add(
+        org.apache.cxf.ws.security.sts.provider.model.wstrust14.ObjectFactory.class
+            .getPackage()
+            .getName());
+
+    ParserConfigurator configurator =
+        parser.configureParser(ctxPath, BaseAuthenticationToken.class.getClassLoader());
+    ByteArrayOutputStream os = new ByteArrayOutputStream();
+
+    JAXBElement<UsernameTokenType> tokenType =
+        new JAXBElement<>(
+            QNameConstants.USERNAME_TOKEN, UsernameTokenType.class, usernameTokenType);
+
+    try {
+      parser.marshal(configurator, tokenType, os);
+      usernameToken = os.toString("UTF-8");
+    } catch (ParserException | UnsupportedEncodingException ex) {
+      LOGGER.info("Unable to parse username token.", ex);
+    }
+
+    return new BaseAuthenticationToken(null, usernameToken);
+  }
+
+  /**
+   * Creates a {@link BaseAuthenticationToken} from a given list of certificates. Uses a {@link
+   * BinarySecurityTokenType} internally to store the certificates.
+   *
+   * @param certs - the user's certificates
+   * @return a BaseAuthenticationToken containing the given certificates
+   */
+  public BaseAuthenticationToken getTokenFromCertificates(X509Certificate[] certs) {
+    if (certs == null || certs.length == 0) {
+      return null;
+    }
+
+    byte[] certBytes = null;
+    try {
+      certBytes = getCertBytes(certs);
+    } catch (WSSecurityException e) {
+      LOGGER.debug("Unable to convert PKI certs to byte array.", e);
+    }
+
+    if (certBytes == null) {
+      return null;
+    }
+
```

Comment:
Now that we don't have the PKIAuthenticationToken and PKITokenValidator, we need to use a BinarySecurityTokenType to store the cert info to pass into the STS so it can be validated correctly. This is what we were doing in the [WssPKIHandler](https://github.com/codice/ddf/pull/4620/files#diff-7c9d98cffca971af09df1e86ff954f19L46)